src/template/wasm2c.declarations.c

```diff
@@ -613,3 +619,21 @@ DEFINE_TABLE_FILL(externref)
 #define FUNC_TYPE_EXTERN_T(x) const char x[]
 #define FUNC_TYPE_T(x) static const char x[]
 #endif
+
+#if (__STDC_VERSION__ < 201112L) && !defined(static_assert)
+#define static_assert(X) \
+  extern int(*assertion(void))[!!sizeof(struct { int x : (X) ? 2 : -1; })];
+#endif
+
+#ifdef _MSC_VER
+#define WEAK_FUNC_DECL(func, fallback)                             \
+  __pragma(comment(linker, "/alternatename:" #func "=" #fallback)) \
+                                                                   \
+      void                                                         \
+      fallback(void** instance_ptr, void* tail_call_stack,         \
+               wasm_rt_tailcallee_t* next)
+#else
+#define WEAK_FUNC_DECL(func, fallback)                                        \
+  __attribute__((weak)) void func(void** instance_ptr, void* tail_call_stack, \
+                                  wasm_rt_tailcallee_t* next)
```

Comment:
This version seems to completely ignore `fallback`.. how does that work?

Reply:
i think the fallback is just an alternate way to implement weak symbols in MSVC? IIUC, they aren't used in the non MSVC environments as weak symbols are available.

Reply:
I guess I don't understand how that fallback function is used on MSVC but a weak symbol is used otherwise.    Do we also declare the fallback symbol on under MSVC?

Reply:
Yeah, the "fallback" parameter is only used on MSVC.

On GCC/clang, the "weak import" is implemented by defining a stub version of the import under its real name, and marking it as a weak symbol so it will be overridden by the real import if it exists.

On MSVC, it's implemented by defining the stub version of the import under the "fallback" name, and then telling the linker to use the fallback as a substitute for the real import if the real import doesn't exist.